DOCS-5432 add section about oauth access token (#786)
clerk/clerk-docs: docs/authentication/social-connections/oauth.mdx

Clerk makes it easy to add social connection capabilities to your application. Social connections are designed to simplify the sign-up and sign-in process for the end-user, resulting in higher conversion rates, a streamlined user data collection flow, and an overall better user experience.

> Social connections provide better security than passwords and other long-lived knowledge-based secrets. With social connections, users gain frictionless access to your application by using their existing credentials from an OAuth provider (Google, Facebook, X/Twitter, etc.) without having to go through complicated registration flows.

Author: I have no clue who wrote this LMAO I can remove it though o.O

This example is written for Next.js App Router, but is supported by any React meta framework, such as Remix or Gatsby.

```tsx filename="app/api/notion/route.tsx"
import { auth, clerkClient } from "@clerk/nextjs";
import { NextResponse } from "next/server";

export async function GET() {
  const { userId } = auth();

  if (!userId) {
    return NextResponse.json({ message: "User not found" });
  }

  // Get the OAuth access token for the user
  const provider = "oauth_notion";

  const clerkResponse = await clerkClient.users.getUserOauthAccessToken(
    userId,
    provider
  );

  const accessToken = clerkResponse[0].token;

  // Fetch the user data from the Notion API
  // This endpoint fetches a list of users
  // https://developers.notion.com/reference/get-users
  const notionUrl = "https://api.notion.com/v1/users";

  const notionResponse = await fetch(notionUrl, {
    headers: {
      Authorization: `Bearer ${accessToken}`,
      "Notion-Version": "2022-06-28",
    },
  });

  // Handle the response from the Notion API
  const notionData = await notionResponse.json();

  return NextResponse.json({ message: notionData });
}

> ```

Reviewer: 😍

For [Clerk Components](/docs/components/overview), this is handled automatically by Clerk. You only have to pass the [`additionalOAuthScopes`](/docs/components/user/user-profile) prop in the [`<UserProfile/>`](/docs/components/user/user-profile) component or the corresponding [`userProfileProp`](/docs/components/user/user-button) in the [`<UserButton />`](/docs/components/user/user-button) component, with any additional OAuth scope you would like per provider.

## Get an OAuth access token for a social login provider

> To query an OAuth progvider for additional data of your users, you can use the OAuth access token.

Reviewer: ```suggestion
You can use a provider's OAuth access token to access user data from that provider in addition to their data from Clerk.
```

> To retrieve the OAuth access token for a user, use the `getUserOauthAccessToken()` method from the Clerk Backend SDK. Note that this method must be used in a server environment, and cannot be run on the client.

Reviewer: We should link to the page for it here

Author: oh I'm silly I forgot that

> Clerk ensures that the OAuth Access Token will be always fresh so that you don't have to worry about OAuth Refresh Tokens anymore.

Author: added!
